ApollosProject/apollos-apps
# feat: Daily Habit Feature Release

## Motivation
This PR contains the final details required to launch the fully connected Daily Spiritual Habit Experience in its MVP form.

For a full product demo, please check out this [Loom](https://www.loom.com/share/8fdc0b7cf08f420281fa09dff26fa218)

## How to Test
You need to make sure that you have your Feature set up correctly in the dB. There is nothing inside of Staging nor production right now for this, but this configuration _should_ get you set up quickly

There is an API dependency on these changes: please make sure to use this [associated API branch](https://github.com/ApollosProject/apollos-cluster/pull/271) for it all to work as expected

1. Inside of the `feature` table, add a new feature:
```
type: **DailyHabitCardList**
apollos_type: **DailyHabitCardListFeature**
church_slug: **apollos_demo**
data: {
  "type": "DailyHabitCardList",
  "title": "Daily Spiritual Habits",
  "subtitle": "Boost your daily habit with a warm-up today.",
  "algorithms": [
    {
      "type": "HABIT_GRATITUDE"
    },
    {
      "type": "HABIT_PRAYER"
    },
    {
      "type": "HABIT_READING",
      "arguments": {
        "categoryIDs": [
          "5bf609c1-5689-421f-a27a-118d5b7006a7"
        ]
      }
    }
  ]
}
```

*You might need to update your `categoryIDs` inside of the `data`
**HABIT_READING requires that the Content Item includes a Scripture Feature inside of it. If you notice that it's not working as expected, make sure that the `categoryIDs` that are being chosen render Content Items that include at least 1 Scripture Feature

2. Add the new Feature to a Feature Feed (Home is the easiest)

Now that you're ready to test everything, go ahead and pop around on each of the different experiences. The "intended" use case would be to hit Gratitude then Prayer then Reading, but they aren't required to go in that order.

## What makes an Experience "Complete"?
**Gratitude:** posting a Gratitude
**Prayer:** praying for another user
**Read:** open up a Scripture Card on today's Daily Article
